src/Foundation/NSUrlCredential.cs

```diff
@@ -15,20 +15,29 @@ namespace Foundation {
 
 	public partial class NSUrlCredential {
 		public NSUrlCredential (SecIdentity identity, SecCertificate [] certificates, NSUrlCredentialPersistence persistence)
-			: this (identity.Handle, NSArray.FromNativeObjects (certificates).Handle, persistence)
```

Comment:
This could have been another solution:

```cs
	: this (identity.DangerousRetain ().DangerousAutorelease ().Handle, NSArray.FromNativeObjects (certificates).DangerousRetain ().DangerousAutorelease ().Handle, persistance)
```

That said, I think yours is better.